File: sdk/src/main/java/com/meniga/sdk/webservices/requests/StartRealmSync.java
package com.meniga.sdk.webservices.requests;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

/**
 * Copyright 2017 Meniga Iceland Inc.
 * Created by agustk on 4.4.2018.
 */
public class StartRealmSync extends QueryRequestObject {
    public transient long realmUserId;
    public long waitForCompleteMilliseconds;

    @Override
    public long getValueHash() {
        int hash = (int) (waitForCompleteMilliseconds ^ (waitForCompleteMilliseconds >>> 32));
        hash += (int) (realmUserId ^ (realmUserId >>> 32));
        return hash;
    }

    @Override
    public Map<String, String> toQueryMap() {
        Map<String, String> query = new HashMap<>();
        query.put("realmUserId", Long.toString(realmUserId));
        return query;
    }
}

File: sdk/src/main/java/com/meniga/sdk/webservices/requests/StartRealmSync.kt
package com.meniga.sdk.webservices.requests

/**
 * Copyright 2018 Meniga Iceland Inc.
 * Created by agustk on 5.4.2018.
 */
data class StartRealmSync(
        @JvmField @Transient var realmUserId: Long = 0,
        @JvmField var waitForCompleteMilliseconds: Long = 0,
        @JvmField var sessionToken: String? = null
) : QueryRequestObject() {
    override fun getValueHash(): Long = hashCode().toLong()
}
